src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -202,7 +366,8 @@ private static async Task<VersionStamp> GetDocumentSemanticVersionAsync(Document
             if (!_documentMappingService.TryMapToProjectedDocumentRange(codeDocument, razorRange, out var csharpRange) &&
                 !TryGetMinimalCSharpRange(codeDocument, razorRange, out csharpRange))
             {
-                return SemanticRangeResponse.Default;
+                // This probably means there's no C# in the range.
+                return new SemanticRangeResponse(null, IsCSharpFinalized: true);
```

Comment:
Nit: whenever passing null it's helpful to explicitly name the parameter to provide context.